File: ui/hospital-system-ui/src/types/supplier.ts
export interface SupplierResponse {
  id: number
  companyName: string
  taxNumber: string
  taxOffice: string | null
  bankName: string | null
  bankBranch: string | null
  iban: string | null
  contactName: string | null
  phone: string | null
  email: string | null
  address: string | null
  createdAt: string
  updatedAt: string
}

export interface CreateSupplierRequest {
  companyName: string
  taxNumber: string
  taxOffice?: string
  bankName?: string
  bankBranch?: string
  iban?: string
  contactName?: string
  phone?: string
  email?: string
  address?: string
}

export interface UpdateSupplierRequest {
  companyName?: string
  taxNumber?: string
  taxOffice?: string
  bankName?: string
  bankBranch?: string
  iban?: string
  contactName?: string
  phone?: string
  email?: string
  address?: string
}

File: server/src/main/java/com/hospitaltender/server/entity/Supplier.java
package com.hospitaltender.server.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "suppliers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Supplier extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "company_name", nullable = false)
    private String companyName;

    @Column(name = "tax_number", length = 20, unique = true, nullable = false)
    private String taxNumber;

    @Column(name = "tax_office", length = 100)
    private String taxOffice;

    // Bank Information
    @Column(name = "bank_name", length = 100)
    private String bankName;

    @Column(name = "bank_branch", length = 100)
    private String bankBranch;

    @Column(name = "iban", length = 34)
    private String iban;

    // Contact Information
    @Column(name = "contact_name", length = 100)
    private String contactName;

    @Column(name = "phone", length = 20)
    private String phone;

    @Column(name = "email", length = 100)
    private String email;

    @Column(name = "address", columnDefinition = "TEXT")
    private String address;

    @Column(name = "is_blacklisted")
    private Boolean isBlacklisted = false;
}
